### PR TITLE
Simplify base TestCases

### PR DESCRIPTION
**Pull request by 1ucian0**
Removing the complexity in the TestCases class hierarchy. Additionally, removing the terra mentions.

### CODE REVIEW

File: test/utils/base.py [hunk 1]
```diff
@@ -65,11 +65,97 @@ class BaseTestCase(unittest.TestCase):
 
 
 @enforce_subclasses_call(["setUp", "setUpClass", "tearDown", "tearDownClass"])
-class BaseQiskitTestCase(BaseTestCase):
-    """Additions for test cases for all Qiskit-family packages.
+class QiskitTestCase(BaseTestCase):
+    """Additions for test cases for Qiskit."""
```
Comment (ElePT) [inline test/utils/base.py]:
```suggestion
    """Additions for Qiskit test cases."""
```

File: test/utils/base.py [hunk 2]
```diff
@@ -165,112 +258,10 @@ def set_parallel_env(name, value):
...
-            # Caused by internal scikit-learn scipy usage
-            r"The 'sym_pos' keyword is deprecated and should be replaced by using",
-        ]
-        for msg in allow_DeprecationWarning_message:
-            warnings.filterwarnings("default", category=DeprecationWarning, message=msg)
-
-
 class FullQiskitTestCase(QiskitTestCase):
-    """Terra-specific further additions for test cases, if ``testtools`` is available.
+    """Qiskit further additions for test cases, if ``testtools`` is available.
```
Comment (ElePT) [inline test/utils/base.py]:
```suggestion
    """further additions for Qiskit test cases, if ``testtools`` is available.
```

**Review by ElePT — COMMENTED** [review]
I think it makes sense to remove the base class, I added two small wording suggestions, but LGTM.

**Comment by 1ucian0** [comment]
>  I added two small wording suggestions

accepting both.

**Review by ElePT — APPROVED** [review]
LGTM!

**Pull request merged**